src/ompl/geometric/planners/experience/LRR.h

```diff
@@ -0,0 +1,205 @@
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dave Coleman */
+
+#ifndef OMPL_GEOMETRIC_PLANNERS_EXPERIENCE_LRR_
+#define OMPL_GEOMETRIC_PLANNERS_EXPERIENCE_LRR_
+
+#include "ompl/geometric/planners/PlannerIncludes.h"
+#include "ompl/geometric/PathGeometric.h"
+#include "ompl/geometric/PathSimplifier.h"
+#include "ompl/datastructures/NearestNeighbors.h"
+
+namespace ompl
+{
+    namespace tools
+    {
+        OMPL_CLASS_FORWARD(LightningDB);
+    }
+
+    namespace geometric
+    {
+
+        /// @cond IGNORE
+        /** \brief Forward declaration of ompl::base::LRR */
+        OMPL_CLASS_FORWARD(LRR);
+        /// @endcond
+
+        /** \class ompl::base::LRRPtr
+            \brief A boost shared pointer wrapper for ompl::base::LRR */
+
+        /**
+           @anchor LRR - Lightning Retrieve Repair
+           @par Short description
+           LRR is a experienced-based motion planner that recalls from a database of
+           previous actions the most similar one to the current planning problem and attempts to repair it
+           @par External documentation
+           Berenson, Dmitry, Pieter Abbeel, and Ken Goldberg: A robot path planning framework that learns from experience, in <em>Robotics and Automation (ICRA), 2012 IEEE International Conference on. IEEE</em>, 2012.
+           <a href="http://users.wpi.edu/~dberenson/lightning.pdf">[PDF]</a>
+        */
+
+        /** \brief The Lightning Framework's Retrieve-Repair component */
+        class LRR : public base::Planner
+        {
+        public:
+
+            /** \brief Constructor */
+            LRR(const base::SpaceInformationPtr &si, const ompl::tools::LightningDBPtr &experienceDB);
+
+            virtual ~LRR(void);
+
+            /** \brief Get information about the exploration data structure the planning from scratch motion planner used. */
+            virtual void getPlannerData(base::PlannerData &data) const;
+
+            /**
+             *  \brief Get debug information about the top recalled paths that were chosen for further filtering
+             *  \return data - vector of PlannerData objects that each hold a single path
+             */
+            const std::vector<base::PlannerDataPtr>& getLastRecalledNearestPaths() const;
+
+            /**
+             *  \brief Get debug information about the top recalled paths that were chosen for further filtering
+             *  \return chosenID - the index of the PlannerData object that was chosen for repair
+             */
+            const std::size_t& getLastRecalledNearestPathChosen() const;
+
+            /**
+             * \brief Get the chosen path used from database for repair
+             * \return PlannerData of chosen path
+             */
+            ompl::base::PlannerDataPtr getChosenRecallPath() const;
```

Comment:
Actually, yes, Lightning.cpp needs access to that function for solving. Moved back to public.